ksql-serde/src/main/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializer.java

```diff
@@ -90,6 +98,18 @@ public void configure(final Map<String, ?> map, final boolean b) {
     }
   }
 
+  private Schema getNonNullSchema(Schema schema) {
+    if (schema.getType() == Schema.Type.UNION) {
+      List<Schema> schemaList = schema.getTypes();
+      for (Schema innerSchema: schemaList) {
+        if (innerSchema.getType() != Schema.Type.NULL) {
+          return innerSchema;
+        }
+      }
+    }
+    throw new KsqlException("Field need to have at least one not null type." + schema);
```

Comment:
This should never happen now that all fields are unioned with Null by default right? If should, this should be an `IllegalStateException` with an error message like `"Expecting non-null value or a Union type for " + schema.field()`.